client/dashboard/domains/domain-transfer/index.tsx

```diff
@@ -0,0 +1,292 @@
+import { useSuspenseQuery, useMutation } from '@tanstack/react-query';
+import {
+	Card,
+	CardBody,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	ToggleControl,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useLocale } from '../../app/locale';
+import { domainQuery } from '../../app/queries/domain';
+import { domainLockMutation, domainTransferCodeMutation } from '../../app/queries/domain-transfer';
+import { domainRoute } from '../../app/router/domains';
+import { ActionList } from '../../components/action-list';
+import InlineSupportLink from '../../components/inline-support-link';
+import Notice from '../../components/notice';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import RouterLinkButton from '../../components/router-link-button';
+import { SectionHeader } from '../../components/section-header';
+import { DomainSubtype } from '../../data/domains';
+import { formatDate } from '../../utils/datetime';
+import SelectIpsTag from './select-ips-tag';
+
+const RESTRICTED_TRANSFER_TLDS = [ 'uk', 'fr', 'ca', 'de', 'jp' ];
+const TRANSFER_LOCK_GRACE_PERIOD_MS = 60 * 24 * 60 * 60 * 1000; // 60 days
+
+export function getTopLevelOfTld( domainName: string ): string {
+	return domainName.substring( domainName.lastIndexOf( '.' ) + 1 );
+}
```

Comment:
This could be exported as a util method.
There is already a dedicated file for that: `client/dashboard/utils/domain.ts`